include/cantera/oneD/StFlow.h

```
@@ -254,6 +255,87 @@ class StFlow : public Domain1D
 
     void fixTemperature(size_t j=npos);
 
+    /** Two-Point control method(approach)
+     * In this method two control points are designated in the 1D domain, and
+     * the value of the temperature at these points is fixed. The values of the control
+     * points are imposed and thus serve as a boundary condition that affects the
+     * solution of the governing equations in the 1D domain. The imposition of fixed
+     * points in the domain means that the original set of governing equations' boundary
+     * conditions would over-determine the problem. Thus, the boundary conditions are changed
+     * to reflect the fact that the control points are serving as internal boundary conditions.
+     *
+     * The imposition of the two internal boundary conditions requires that two other
+     * boundary conditions be changed. The first is the boundary condition for the continuity equation
+     * at the left boundary, which is changed to be a value that is derived from the solution at the
+     * left boundary. The second is the continuity boundary condition at the right boundary, which is
+     * also determined from the flow solution by using the oxidizer axial velocity equation variable to
+     * compute the mass flux at the right boundary.
+     *
+     * This method is based on the work of M. Nishioka, C.K. Law, and T. Takeno (1996) titled
+     * "A Flame-Controlling Continuation Method for Generating S-Curve Responses with
+     * Detailed Chemistry"
+     **/
+
+    //! Returns the temperature at the left control point
+    double leftControlPointTemperature() const {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            return m_tLeft;
+        } else {
+            return NAN;
+        }
+    }
```

I changed these methods to return `NAN`, rather than having undefined behavior with no return value, in the case that two point control is not active / set up. But I'm wondering if they should raise an exception instead.

---

I like the exception, because you should never be requesting this value if you haven't activated the two-point control.

---

I added exceptions checking if the two-point control method was active and if the values were undefined.